### PR TITLE
chore(templates): renders `deployments` for each stage in the initial pipeline manifest

### PR DESCRIPTION
**Pull request by Lou1415926**
This PR:
1. Refactor unit tests in `manifest` package such that those test suites that are actually testing the integration of `NewX()` and `MarshalBinary` are moved to a new test file.
2. Add go-template rendering in to pipeline manifest YAML so that it renders `stage.deployments` if given.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/manifest/initial_manifest_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+			tpl, err := manifest.MarshalBinary()
+			require.NoError(t, err)
+
+			// THEN
+			require.Equal(t, string(wantedBytes), string(tpl))
+		})
+	}
+}
+
+func TestPipelineManifest_InitialManifest_Integration(t *testing.T) {
```
Comment (Lou1415926) [inline internal/pkg/manifest/initial_manifest_integration_test.go:337]: Although GH has hidden all the diffs in this file, this test suite is actually relevant.

File: internal/pkg/template/templates/cicd/pipeline.yml [hunk 2]
```diff
@@ -25,6 +25,12 @@ source:
 stages:{{range .Stages}}
     - # The name of the environment.
       name: {{.Name}}
+      {{- if .Deployments}}{{range $actionName, $action := .Deployments}}
+      deployments:
+        {{$actionName}}:
+          template_path: {{$action.TemplatePath}}
+          template_config: {{$action.TemplateConfig}}
+      {{- end}}{{end}}
```
Comment (efekarakus) [inline internal/pkg/template/templates/cicd/pipeline.yml:33]: ooh given this, what's the reason for having the `env/pipeline.yml` aren't they almost identical?

Comment (Lou1415926) [reply internal/pkg/template/templates/cicd/pipeline.yml:33]: I forgot to delete it 😅  I created that file before I realized I could just use `pipeline.yml`

**Review by efekarakus — APPROVED** [review]

**Review by huanjani — APPROVED** [review]
LGTM!

**Pull request merged**